### PR TITLE
IoT SDK review updates

### PR DESCRIPTION
**Pull request by ericwol-msft**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,11 @@
 - Rename `az_iot_hub_client_properties` to `az_iot_message_properties` and move it from `az_iot_hub_client.h` to `az_iot_common.h`.
 - Remove `az_pair` from `az_iot_message_properties_next()` in favor of individual name and value `az_span` parameters.
 - In `az_result.h`, rename `az_failed()` to `az_result_failed()` and `az_succeeded()` to `az_result_succeeded()`.
+- az_iot_is_success_status() renamed to az_iot_status_succeeded()
+- az_iot_is_retriable_status() renamed to az_iot_status_retriable()
+- az_iot_retry_calc_delay() renamed to az_iot_calculate_retry_delay()
+- az_iot_hub_client_sas_get_password() parameter token_expiration_epoch_time moved to second parameter
+- az_iot_provisioning_client_init() parameter global_device_endpoint renamed to global_device_hostname
```
Comment (ahsonkhan) [inline CHANGELOG.md:21]: nit: Add back-ticks and period.
```suggestion
- `az_iot_is_success_status()` renamed to `az_iot_status_succeeded()`.
- `az_iot_is_retriable_status()` renamed to `az_iot_status_retriable()`.
- `az_iot_retry_calc_delay()` renamed to `az_iot_calculate_retry_delay()`.
- `az_iot_hub_client_sas_get_password()` parameter `token_expiration_epoch_time` moved to second parameter.
- `az_iot_provisioning_client_init()` parameter `global_device_endpoint` renamed to `global_device_hostname`.
```

File: sdk/inc/azure/iot/az_iot_provisioning_client.h [hunk 2]
```diff
@@ -69,7 +69,7 @@ AZ_NODISCARD az_iot_provisioning_client_options az_iot_provisioning_client_optio
  * @brief Initializes an Azure IoT Provisioning Client.
  *
  * @param[in] client The #az_iot_provisioning_client to use for this call.
- * @param[in] global_device_endpoint The global device endpoint.
+ * @param[in] global_device_hostname The global device endpoint.
```
Comment (ahsonkhan) [inline sdk/inc/azure/iot/az_iot_provisioning_client.h:72]: nit: Worth updating the doc comment too?

File: sdk/samples/iot/generate_certificate.ps1 [hunk 3]
```diff
@@ -53,10 +53,10 @@ Write-Output "`nSAMPLE CERTIFICATE GENERATED:"
 Write-Output "Use the following command to set the environment variable for the samples:"
 
 if ($IsWindows) {
-  Write-Output "`n`t`$env:AZ_IOT_DEVICE_X509_CERT_PEM_FILE=$(Resolve-Path device_cert_store.pem)"
+  Write-Output "`n`t`$env:AZ_IOT_DEVICE_X509_CERT_PEM_FILE_PATH=$(Resolve-Path device_cert_store.pem)"
```
Comment (ahsonkhan) [inline sdk/samples/iot/generate_certificate.ps1:56]: Nice catch!

Any way we can make sure this script continues to work?

File: sdk/inc/azure/iot/az_iot_provisioning_client.h [hunk 4]
```diff
@@ -290,7 +290,9 @@ AZ_NODISCARD az_result az_iot_provisioning_client_parse_operation_status(
 
 /**
  * @brief Checks if the status indicates that the service has an authoritative result of the
- * register operation. The operation may have completed in either success or error.
+ * register operation. The operation may have completed in either success or error. Compleated
```
Comment (ahsonkhan) [inline sdk/inc/azure/iot/az_iot_provisioning_client.h:293]:
```suggestion
 * register operation. The operation may have completed in either success or error. Completed

```

File: sdk/inc/azure/iot/az_iot_provisioning_client.h [hunk 5]
```diff
@@ -329,7 +331,7 @@ AZ_NODISCARD az_result az_iot_provisioning_client_register_get_publish_topic(
  * @remark The payload of the MQTT publish message should be empty.
  *
  * @param[in] client The #az_iot_provisioning_client to use for this call.
- * @param[in] register_response The received #az_iot_provisioning_client_register_response response.
+ * @param[in] operation_id The received #az_iot_provisioning_client_register_response response.
```
Comment (ahsonkhan) [inline sdk/inc/azure/iot/az_iot_provisioning_client.h:334]: Doc comment needs update.
```suggestion
 * @param[in] operation_id The received operation id from the #az_iot_provisioning_client_register_response response.
=
```

File: sdk/inc/azure/iot/az_iot_hub_client.h [hunk 6]
```diff
@@ -32,9 +32,7 @@
  */
 enum
 {
-  AZ_HUB_CLIENT_DEFAULT_MQTT_TELEMETRY_QOS = 0,
-  AZ_HUB_CLIENT_DEFAULT_MQTT_TELEMETRY_DUPLICATE = 0,
-  AZ_HUB_CLIENT_DEFAULT_MQTT_TELEMETRY_RETAIN = 1
```
Comment (ahsonkhan) [inline sdk/inc/azure/iot/az_iot_hub_client.h:37]: Add the removal of these named constants to the changelog.

File: sdk/samples/iot/generate_certificate.ps1 (hunk 3 above)
Comment (ericwol-msft) [reply sdk/samples/iot/generate_certificate.ps1:56]: that is how I found it

File: sdk/inc/azure/iot/az_iot_provisioning_client.h (hunk 2 above)
Comment (ericwol-msft) [reply sdk/inc/azure/iot/az_iot_provisioning_client.h:72]: that was already fixed in the PR

File: sdk/inc/azure/iot/az_iot_provisioning_client.h (hunk 5 above)
Comment (ericwol-msft) [reply sdk/inc/azure/iot/az_iot_provisioning_client.h:334]: was already fixed in the PR

File: sdk/inc/azure/iot/az_iot_provisioning_client.h (hunk 2 above)
Comment (ahsonkhan) [inline sdk/inc/azure/iot/az_iot_provisioning_client.h:72]: I meant the actual description of the parameter.
```suggestion
 * @param[in] global_device_hostname The global device hostname.
```

File: sdk/inc/azure/iot/az_iot_provisioning_client.h (hunk 5 above)
Comment (ahsonkhan) [reply sdk/inc/azure/iot/az_iot_provisioning_client.h:334]: I meant the actual **description** of the parameter documentation. See suggested change.

File: sdk/samples/iot/generate_certificate.ps1 (hunk 3 above)
Comment (momuno) [reply sdk/samples/iot/generate_certificate.ps1:56]: May final task for the samples is to review documentation with a test run.  The ps1 script is a new add, so has not received this test yet.

**Pull request merged**